libs/MeshKernel/include/MeshKernel/AveragingInterpolation.hpp

```diff
@@ -104,14 +105,34 @@ namespace meshkernel
                                size_t minNumSamples);
 
         /// @brief Compute interpolation
-        void Compute();
+        void Compute() override;
 
-        /// @brief Get the result values
-        /// @return the results
-        [[nodiscard]] const auto& GetResults() const
-        {
-            return m_results;
-        }
+        /// @brief Gets the interpolation value at a specific node
+        /// @param[in] node The node index
+        /// @return The interpolated value
+        [[nodiscard]] double GetNodeResult(size_t node) const override { return m_nodeResults[node]; }
```

Comment:
Apart from Compute, there's no need to override the remaining virtual methods here. they can be all implemented in the base and inherited.